addons/pos_restaurant/static/src/js/multiprint.js

```diff
@@ -201,13 +202,13 @@ models.Order = models.Order.extend({
     },
     computeChanges: function(categories){
         var current_res = this.build_line_resume();
-        var old_res     = this.saved_resume || {};
+        var old_res     = this.saved_resume || [];
         var json        = this.export_as_JSON();
         var add = [];
         var rem = [];
         var line_hash;
 
-        for ( line_hash in current_res) {
+        for (line_hash = 0; line_hash < current_res.length; line_hash++) {
             var curr = current_res[line_hash];
             var old  = old_res[line_hash];
```

Comment:
this whole thing is broken now isn't it? You don't want to compare elements with the same index in both lists, you want to compare items with the same line_hash. Try to rename the line_hash variable to i or something as well. It'll make things clearer, because it's not a line_hash anymore.